examples/sphere/shallow_water.jl

```diff
@@ -171,6 +184,76 @@ if test_name == rossby_haurwitz_test_name
             Geometry.UVVector(uλ, uϕ),
             local_geometry,
         )
+        return (h = h, u = u)
+    end
+elseif test_name == steady_state_compact_test_name
+    y0 = map(Fields.local_geometry_field(space)) do local_geometry
+        coord = local_geometry.coordinates
+
+        ϕ = coord.lat
+        λ = coord.long
+
+        if α == 0.0
+            ϕprime = ϕ
+            λprime = λ
+        else
+            ϕprime = asind(sind(ϕ) * cosd(α) - cosd(ϕ) * cosd(λ) * sind(α))
+            λprime = asind(sind(λ) * cosd(ϕ) / cosd(ϕprime))
+
+            # Temporary angle to ensure λprime is in the right quadrant
+            λcond = cosd(α) * cosd(λ) * cosd(ϕ) + sind(α) * sind(ϕ)
+
+            # If λprime is not in the right quadrant, adjust
+            if λcond < 0.0
+                λprime = -λprime -180.0
+            end
+            if λprime < -180.0
+                λprime += 360.0
+            end
+        end
+
+        # Set auxiliary function needed for initial state of velocity field
+        b(x) = x ≤ 0.0 ? 0.0 : exp(-x^(-1))
+
+        x(ϕprime) = xₑ * (ϕprime - ϕᵦ) / (ϕₑ - ϕᵦ)
+        uλprime(ϕprime) =
+            u0 * b(x(ϕprime)) * b(xₑ - x(ϕprime)) * exp(4.0 / xₑ)
+        uϕprime = 0.0
+
+        # Set integral needed for height initial state
+        hᵢₙₜ(γ) = abs(γ) < 90.0 ? (2 * Ω * sind(γ) + uλprime(γ) * tand(γ) / R) * uλprime(γ) : 0.0
```

Comment:
This is what I see

<img width="875" alt="Screen Shot 2021-11-03 at 3 08 38 PM" src="https://user-images.githubusercontent.com/1880641/140200113-8aca837c-9799-4d85-b6ca-3dff97583e81.png">


cc @simonbyrne

Reply:
Oh, interesting. I got the variable name from what we were using in a [similar test case in ClimateMachine.jl](https://github.com/CliMA/ClimateMachine.jl/blob/46b0828569134b8d7ceb68559e76f1abf15968f8/test/Numerics/DGMethods/compressible_navier_stokes_equations/sphere/benchmark_barotropic_instability.jl#L79)

Reply:
We can rename it, of course

Reply:
it renders correctly on mine, but looks a bit funny. lets just call it `h_int`